File: api/calculator.js
module.exports = (req, res) => {
  const body = req.body || {};
  const op = (body.operation || "add").toLowerCase();
  const a = parseFloat(body.a || 0);
  const b = parseFloat(body.b || 0);
  let result = 0;
  if (op === "add") result = a + b;
  else if (op === "subtract") result = a - b;
  else if (op === "multiply") result = a * b;
  else if (op === "divide") result = b !== 0 ? a / b : 0;
  else if (op === "power") result = Math.pow(a, b);
  else if (op === "square_root") result = Math.sqrt(a);
  else if (op === "percentage") result = (a * b) / 100;
  else if (op === "modulus") result = b !== 0 ? a % b : 0;
  res.status(200).json({ operation: op, a, b, result });
};

File: utils/calculator.py
import math

def add(a, b):
    return a + b

def subtract(a, b):
    return a - b

def multiply(a, b):
    return a * b

def divide(a, b):
    if b == 0:
        raise ZeroDivisionError('Cannot divide by zero')
    return a / b

def power(a, b):
    return a ** b

def square_root(a):
    if a < 0:
        raise ValueError('Cannot calculate square root of negative number')
    return math.sqrt(a)

def factorial(n):
    if n < 0:
        raise ValueError('Cannot calculate factorial of negative number')
    elif n == 0 or n == 1:
        return 1
    else:
        return math.factorial(n)

def percentage(a, b):
    if b == 0:
        raise ValueError('Cannot calculate percentage with zero divisor')
    return (a / b) * 100

def modulus(a, b):
    if b == 0:
        raise ValueError('Cannot calculate modulus with zero divisor')
    return a % b